framework/origin_trials_client.py

```diff
@@ -0,0 +1,52 @@
+# Copyright 2023 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from dataclasses import asdict
+from typing import Any
+import requests
+
+from framework import secrets
+from internals.data_types import OriginTrialInfo
+import settings
+
+
+
+# Data type for information if an error occurred. Error code + message.
+ERROR_INFO_TYPE = tuple[int, str]|None
+
+
+def get_trials_list() -> tuple[list[dict[str, Any]], ERROR_INFO_TYPE]:
+  """Get a list of all origin trials."""
+  key = secrets.get_ot_api_key()
+  # Return an empty list if no API key is found.
+  if key == None:
+    return [], None
+
+  try:
+    response = requests.get(
+        f'{settings.OT_API_URL}/v1/trials',
+        params={'prettyPrint': 'false', 'key': key})
+    response.raise_for_status()
+  except requests.exceptions.HTTPError:
```

Comment:
I want to make sure I'm doing this right - should I not be passing the error info in the return value and instead be returning an actual error? Or am I raising the error in the `get_trials_list()` function and then catching it in the `origin_trials_api.do_get()` to raise it again there?

Reply:
You'll do this part `Or am I raising the error in the get_trials_list() function and then catching it in the origin_trials_api.do_get()`. 

Then in do_get(), it will catch that specific exception and call the right abort:

```py
try:
  trials_list = origin_trials_client.get_trials_list()
except ExceptionType1: # the http one
  self.abort(500, 'Error obtaining origin trial data from API')
except ExceptionType2: # The valueerror one
  self.abort(500, 'Malformed response from origin trials API')
```